Include/internal/pycore_runtime.h

```diff
@@ -84,6 +84,7 @@ typedef struct pyruntimestate {
            If that becomes a problem later then we can adjust, e.g. by
            using a Python int. */
         int64_t next_id;
+        int allow_new;
```

Comment:
FWIW, in the long term I think we'll end up with `_PyRuntimeState.ready`, rendering `_PyRuntimeState.interpreters.allow_new` superfluous.  However, I'm fine with keeping it focused on our immediate needs here.  It won't be hard to switch it over later.